src/App.jsx

```diff
@@ -14,33 +15,94 @@ export const goodsFromServer = [
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const NOT_SORTED = '';
+const SORTED_BY_NAME = 'name';
+const SORTED_BY_LENGTH = 'length';
+const REVERSED = 'reversed';
+const NOT_REVERSED = 'not reversed';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goodsFrom, sort, revStatus) {
+  const goods = [...goodsFrom];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sort) {
+    goods.sort((good1, good2) => {
+      switch (sort) {
+        case SORTED_BY_NAME:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORTED_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (revStatus === REVERSED) {
+    return goods.reverse();
+  }
+
+  return goods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortBy, setSortBy] = useState(NOT_SORTED);
+  const [reverseStatus, setReverseStatus] = useState(NOT_REVERSED);
+
+  const sortedGoods = getPreparedGoods(goodsFromServer, sortBy, reverseStatus);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortBy(SORTED_BY_NAME)}
+          type="button"
+          className={`button is-info ${sortBy !== SORTED_BY_NAME ? 'is-light' : ''}`}
+        >
+          Sort alphabetically
+        </button>
+        <button
+          onClick={() => setSortBy(SORTED_BY_LENGTH)}
+          type="button"
+          className={`button is-success ${sortBy !== SORTED_BY_LENGTH ? 'is-light' : ''}`}
+        >
+          Sort by length
+        </button>
+        <button
+          onClick={() =>
+            reverseStatus === REVERSED
+              ? setReverseStatus(NOT_REVERSED)
+              : setReverseStatus(REVERSED)
+          }
+          type="button"
+          className={`button is-warning ${reverseStatus !== REVERSED ? 'is-light' : ''}`}
+        >
+          Reverse
+        </button>
+        {reverseStatus !== NOT_REVERSED || sortBy !== NOT_SORTED ? (
+          <button
+            onClick={() => {
+              setReverseStatus(NOT_REVERSED);
+              setSortBy(NOT_SORTED);
+            }}
+            type="button"
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        ) : (
+          ''
+        )}
```

Comment:
```suggestion
        {reverseStatus !== NOT_REVERSED || sortBy !== NOT_SORTED && (
          <button
            onClick={() => {
              setReverseStatus(NOT_REVERSED);
              setSortBy(NOT_SORTED);
            }}
            type="button"
            className="button is-danger is-light"
          >
            Reset
          </button>
        )}
```
You can write like this or return null